src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,8 +1,66 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private static final int INDEX_OF_NAME = 0;
+    private static final int INDEX_OF_NUM = 1;
+    private static final String VARIABLE_BUY = "buy";
+    private static final String VARIABLE_SUPPLY = "supply";
+    private static final String VARIABLE_RESULT = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        writeToFile(fromFileName, toFileName);
+    }
+
+    private void writeToFile(String fromFileName, String toFileName) {
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
+            String allData = getReport(readInfoFromFile(fromFileName));
+            writer.write(allData);
+        } catch (IOException e) {
+            throw new RuntimeException("Can't write data to file: " + toFileName, e);
+        }
+    }
```

Comment:
```suggestion
    private void writeToFile(String report, String toFileName) {
        try (BufferedWriter writer = new BufferedWriter(new FileWriter(toFileName))) {
            writer.write(report);
        } catch (IOException e) {
            throw new RuntimeException("Can't write data to file: " + toFileName, e);
        }
    }
```
давай зробимо приватні методи незалежними один від одного і викликатимемо їх в методі getStatistic() створюючи змінні та передаючи їх в настумний метод step-by-step